lib/ae_mdw/activities.ex

```diff
@@ -152,8 +167,55 @@ defmodule AeMdw.Activities do
     end
   end
 
-  defp build_gens_stream(_state, _direction, _account_pk, _gen_scope, _cursor) do
-    []
+  defp build_gens_stream(gen_activities, direction) do
+    Stream.flat_map(gen_activities, fn [{height, _data} | _rest] = chunk ->
+      gen_events =
+        chunk
+        |> Enum.sort()
+        |> Enum.with_index()
+        |> Enum.map(fn {{^height, data}, local_idx} -> {{height, -1, local_idx}, data} end)
+
+      if direction == :forward do
+        gen_events
+      else
+        Enum.reverse(gen_events)
```

Comment:
I mean it could be avoided if the `Enum.sort` is `:desc` for `:backward`

Reply:
I could do that, but I would also have to index these records in a reverse order. Honestly, I don't think it's the total amount of code I would need to add for sorting a very minimal amount of records

Reply:
Once it was mentioned that this endpoint might become the main one to be used it could be worthy but it's ok. A more important question would be if the first sorting by `Enum.sort()` is needed because `data` from each `chunk` element is a `{:int_transfer, account_pk, kind, ref_txi}` where `account_pk` is repeated and `kind` is already sorted by `Enum.map(@kinds,` and `ref_txi` is a constant.

Reply:
That's a good point. Removed it now, no longer sorting the generation int transfers since they are going to be at most 6 and already ordered by kind

Reply:
Thank you 🙏